src/Agent.Listener/Configuration/LinuxServiceControlManager.cs

```diff
@@ -1,14 +1,281 @@
 using System;
 using System.Collections.Generic;
-using System.Threading.Tasks;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
 
 namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
 {
     public class LinuxServiceControlManager : ServiceControlManager
     {
-        public override Task ConfigureServiceAsync(AgentSettings settings, Dictionary<string, string> args, bool enforceSupplied)
+        // This is the name you would see when you do `systemctl list-units | grep vsts`
+        private const string ServiceNamePattern = "vsts.agent.{0}.{1}.service";
+        private const string ServiceDisplayNamePattern = "VSTS Agent ({0}.{1})";
+
+        private const string SystemdPathPrefix = "/etc/systemd/system";
+        private const string InitFileCommandLocation = "/proc/1/comm";
+
+        private const int MaxUserNameLength = 32;
+        private const string VstsAgentServiceTemplate = "vsts.agent.service.template";
+
+        public override void ConfigureService(
+            AgentSettings settings,
+            Dictionary<string, string> args,
+            bool enforceSupplied)
+        {
+            Trace.Info(nameof(ConfigureService));
+
+            CalculateServiceName(settings, ServiceNamePattern, ServiceDisplayNamePattern);
+
+            if (!CheckIfSystemdExists())
```

Comment:
Are we starting with just systemd?  Is it possible to install/apt-get/whatever and get systemd on ubuntu 14.04?  The reason I ask is there's currently too many issues with core CLR on ubuntu 15.10 (I finally gave up recently - just not ready).  I expect that to get resolved before they ship but I'm wondering how we evaluate this work in the short term.  Cent OS?